File: weightloss.nim
import os
import strUtils
import strformat

const start = 118
var appname = extractFilename(getAppFilename())

proc print_usage = 
  echo fmt"usage: {appname} <value> <lb|st|kg>"

proc show_weight(value: float, unit: string) = 
  var
    value_kg: float
    value_st: float
    value_lb: float
    diff:     float
  case unit
  of "kg":
    value_kg = value
  of "lb":
    value_kg = value / 2.2
  of "st":
    value_kg = value * 14.0 / 2.2
  else: 
    echo "unknown unit: ", unit
    echo fmt"usage: {appname} <value> <lb|st|kg>"
    quit 1
  value_st = value_kg * 2.2 / 14
  value_lb = value_kg * 2.2
  diff = start - value_kg
  echo fmt"{value_kg:.2f}kg {value_st:.2f}st {value_lb:.2f}lb (lost {diff:.2f}kg)"

let arguments = commandLineParams()

if len(arguments) < 2:
  echo fmt"usage: {appname} <value> <lb|st|kg>"
  quit 0

show_weight(parseFloat(arguments[0]), arguments[1])

File: weightloss.cpp
#include <iostream>
#include <string>
#include <set>

namespace ERR {
    const int BAD_UNIT = 1;
    const int BAD_USAGE = 2;
}

const float start = 118;

void print_usage(){
    std::cerr << "usage: weightloss <value> <lb|st|kg>"
                  << std::endl;
}

int show_weight(float value, std::string unit){
    float value_kg;
    if (unit == "kg")
    {
        value_kg = value;
    } else if (unit == "lb")
    {
        value_kg = value * 2.2;
    } else if (unit == "st")
    {
        value_kg = value * 2.2 / 14;
    } else {
        std::cout << "Bad unit '" << unit << "'. "
                  << "Use kg, lb, or st."
                  << std::endl;
        return ERR::BAD_UNIT;
    }
    float value_lb = value_kg * 2.2;
    float value_st = value_lb / 14;
    float diff = start - value_kg;
    std::cout.precision(4);
    std::cout << value_kg << "kg "
              << value_st << "st "
              << value_lb << "lb "
              << "(lost " << diff << "kg)"
              << std::endl;
    return 0;
}

int main(int argc, const char** argv) {
    if(argc < 3){
        print_usage();
        return ERR::BAD_USAGE;
    }
    float value = std::stof(argv[1]);
    return show_weight(value, argv[2]);
}
